### PR TITLE
Enable chaining for middlewares and afterwares in the network interface - Issue 564

### PR DESCRIPTION
**Pull request by oricordeau**
TODO:

- [ ] If this PR is a new feature, reference an issue where a consensus about the design was reached (not necessary for small changes)
- [x] Make sure all of the significant new logic is covered by tests
- [x] Rebase your changes on master so that they can be merged easily
- [x] Make sure all tests and linter rules pass
- [x] Update CHANGELOG.md with your change
- [x] Add your name and email to the AUTHORS file (optional)
- [x] If this was a change that affects the external API, update the docs and post a link to the PR in the discussion

### CODE REVIEW

**Comment by oricordeau** [comment]
Hi,
I'm trying to solve #564. Have a look at this PR and tell me what you think.
Remarks :
1. If I'm not mistaking, it should also possible to chain use() and useAfter() in the same call. I might need to also add a test for that. Right now, my patch only tests use(...).use(...) and useAfter(...).useAfter(...). Please tell me if such test is relevant.
2. As this PR adds a new feature, I need to update the docs. I will, but first I'd like to know if the PR looks good.

Regards

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 Expect active development and potentially significant breaking changes in the `0.x` track. We'll try to be diligent about releasing a `1.0` version in a timely fashion (ideally within 3 to 6 months), to signal the start of a more stable API.
 
 ### vNext
+- **new Feature**: In network interface, enable chaining the `use` and `useAfter` functions, respectively. [PR #860](https://github.com/apollostack/apollo-client/pull/860) [Issue #564](https://github.com/apollostack/apollo-client/issues/564)
```
Comment (helfer) [inline CHANGELOG.md]: Can we simplify this to "Enable chaining of `use` and `useAfter` function calls in network interface"?

**Comment by oricordeau** [comment]
Just added a new commit that includes the require change + two new tests for testing use(...).useAfter() and useAfter().use() chaining.
I will now work on a PR on the docs. Stay tuned ...

**Comment by oricordeau** [comment]
Here is a link to a PR on apollostack/core-docs which documents this feature :
https://github.com/apollostack/core-docs/pull/218

**Comment by oricordeau** [comment]
For me, this PR is ready to be merged but tell me if some changes need to be done.

**Comment by helfer** [comment]
Looks good to me. Thanks a lot @oricordeau !

**Pull request merged**